tollway/vehicle.py

```diff
@@ -0,0 +1,102 @@
+from random import choice
+from collections import deque
+
+import requests
+from bs4 import BeautifulSoup, Tag
+from faker import Faker
+from faker_vehicle import VehicleProvider
+from us.states import STATES_AND_TERRITORIES
+
+
+STATE_NAMES = [state.name for state in STATES_AND_TERRITORIES]
+TOLLWAYS_URL = "https://en.wikipedia.org/wiki/List_of_toll_roads_in_the_United_States"
+
+
+def create_vehicle(fake: Faker) -> dict:
+    
+    vehicle = {f"{key.lower()}": value for key, value in fake.vehicle_object().items()}
+    vehicle["license_plate"] = fake.license_plate()
+    vehicle["vin"] = fake.vin()
+    vehicle["state"] = choice(STATES_AND_TERRITORIES).abbr
+    vehicle["primary_color"] = fake.color_name()
+
+    return vehicle
+
+
+def _num_of_tables_between_states(soup: BeautifulSoup) -> dict:
+    states = {}
+    table_count_between_states = 0
+    state_name = ""
+    for row in soup.contents[0]:
+        if not isinstance(row, Tag):
+            continue
+
+        if isinstance(row, Tag) and row.name == "h2" and row.span.string in STATE_NAMES:
+            state_name = row.span.string
+            table_count_between_states = 0
+
+        if isinstance(row, Tag) and row.name == "table" and row["class"][0] == "wikitable" and state_name != "":
+            table_count_between_states += 1
+            states[state_name] = table_count_between_states
+
+    return states
+
+
+def _state_tollways(soup: BeautifulSoup) -> list:
+    names = []
+
+    for table in soup.find_all("table", class_="wikitable"):
+        table_rows = table.tbody.find_all("tr")[1:]
+        state_tollways = []
+        for table_row in table_rows:
+            for tag in table_row.find_all("a"):
+                if tag.string is None:
+                    continue
+                state_tollways.append(tag.string)
+                break
+        names.append(state_tollways)
+    
+    return names
+
+
+def get_tollways() -> dict:
+    tollways_html = requests.get(TOLLWAYS_URL)
+    wikipedia_soup = BeautifulSoup(markup=tollways_html.text, features="html.parser")
+
+    tollway_content = wikipedia_soup.find("div", class_="mw-parser-output")
+    tollway_subsection = BeautifulSoup(markup=str(tollway_content), features="html.parser")
+
+    num_of_tables_between_states = _num_of_tables_between_states(soup=tollway_subsection)
+    state_tollways = deque(_state_tollways(soup=tollway_subsection))
+
+    united_states_tollways = {}
+    for state_name, table_count in num_of_tables_between_states.items():
+
+        merge_tollway_tables = []
+        for _ in range(table_count):
+            merge_tollway_tables.extend(state_tollways.popleft())
+
+        united_states_tollways[state_name] = set(merge_tollway_tables)
+        merge_tollway_tables = []
+
+    return united_states_tollways
+
+
+def create_tollway(tollways: dict) -> [str, str]:
+    state = choice([key for key in tollways.keys()])
```

Comment:
can you just give choice() tollways.keys(), seems the listcomp does not add any value

Reply:
I replaced with list() cast, when I removed the listcomp was receiving type error saying dict_keys object is not subscriptable

Reply:
do you need the cast? keys() is an iterable so that probably just works out of the box?

Reply:
also `[str, str]` type hint issue, should this be `dict[str, str]`?

Reply:
it is returning two strings which is what I was trying to communicate with `[str, str]` however that syntax might be incorrect and instead communicate a list with two strings is expected

Reply:
this is what the output/return looks like in it's current state

<img width="511" alt="Screenshot 2023-11-07 at 7 32 06 AM" src="https://github.com/BenGriffith/tollway-traffic/assets/6089155/bdd458b4-becd-4ae9-ad48-8c61d94ccd10">

Reply:
this is what the return/output looks like in it's current state

<img width="511" alt="Screenshot 2023-11-07 at 7 33 07 AM" src="https://github.com/BenGriffith/tollway-traffic/assets/6089155/b84d8117-970a-4136-b9f2-8da17b9d6199">

Reply:
so maybe I need to use parens to denote tuple instead of brackets

Reply:
thanks for confirming that, then it would be `tuple[str, str]:`